Tasks/Common/npm-common/npmregistry.ts

```diff
@@ -36,6 +40,12 @@ export class NpmRegistry implements INpmRegistry {
 
         try {
             url = NormalizeRegistry(tl.getEndpointUrl(endpointId, false));
+            if (endpointAuth.scheme === 'Token' &&
+                URL.parse(url).hostname.toUpperCase().endsWith('.VISUALSTUDIO.COM')){
```

Comment:
I really don't like this. I suggest renaming the auth type of the endpoint back to 'external VSTS/TFS', and adding a third 'authToken' option.

Reply:
I considered that; however, I don't like the additional complexity/choices that a user would now need to make.  The root of the problem, IMO, is that we don't have a JWT generator that has the same capabilities as the PAT generator under security settings.  A JWT _would_ work with _authToken a PAT; however, does not.

Reply:
We have two options:

1. Add a third endpoint type specific to VSTS, so we have Basic, AuthToken (bearer), and VSTS PAT. 
2. Make the current AuthToken to use bearerToken only and update our guidance for current VSTS customers to use the Basic type and put whatever in the username field.

Reply:
2 would break existing users

Reply:
Yes, forgot to add that 1 would make AuthToken bearer only, which would break existing users as well. This seems like the best option to keep existing users working without having to bump the major version.

Reply:
As we discussed, this is the best approach as it it the least impact on the front-end decision process while taking advantage of the information we have to "do the right thing".